dnase/callsnpsMerge.sh

```diff
@@ -163,6 +164,45 @@ bcftools index ${sampleOutdir}/${name}.${mappedgenome}.filtered.vcf.gz
 tabix -p vcf ${sampleOutdir}/${name}.${mappedgenome}.filtered.vcf.gz
 rm -f ${fltvcffiles}
 
+echo
+# Delly does not have a --regions flag to allow for specific regions to be matched.
+# this makes it difficult to parralelize by chromosome like the bcftools call command
+echo "Run Delly"
+date
+
+## Make a place for the temp bam files with qcfail turned off.
+bam_output_qcOK=${TMPDIR}/${name}.${mappedgenome}.QC_OK_bamfile.bam
+
+## Turn off all qcfail flags.
+python ${src}/changeFlags.py ${sampleOutdir}/${name}.${mappedgenome}.bam ${bam_output_qcOK}
+
+samtools index ${bam_output_qcOK}
+
+# If there are not enough valid reads to run delly it returns a non-zero exit code 
+set +e
+## Look for variants (DEL INS DUP INV TRA).
+delly call -t ALL -o ${sampleOutdir}/${name}.${mappedgenome}.delly.bcf -g ${referencefasta} ${bam_output_qcOK}
+# Store the exit code to see if Delly exited correctly
+dellyCode=$?
+set -e
+
+# If delly exited with a 0 code then create the vcf
+if [ $dellyCode -eq 0 ]; then
+  #bcftools index ${sampleOutdir}/${name}.${mappedgenome}.delly.bcf
+  ## Only accept variants that pass the FILTER test.
+  bcftools filter -i 'FILTER="PASS" & PE>10' ${sampleOutdir}/${name}.${mappedgenome}.delly.bcf | bgzip -c -@ $NSLOTS > ${sampleOutdir}/${name}.${mappedgenome}.delly.filtered.vcf.gz
+  bcftools index ${sampleOutdir}/${name}.${mappedgenome}.delly.filtered.vcf.gz
+
+  echo "Merging Delly VCF"
+  # Output a new VCF with the combined bcftools and Delly calls
+  # FIXME: Should the separate vcfs be temporary files or not?
```

Comment:
yes, the final VCF file must be called ${sampleOutdir}/${name}.${mappedgenome}.filtered.vcf.gz

Probably easiest to create the combined VCF in $TMPDIR, then move it into place

Reply:
That is what I had in mind.
I will fix this in the next commit.